main.go

```diff
@@ -15,47 +17,47 @@ func printHelp() {
 		" gosplat [input dir/file] , runs directory against model\n")
 }
 
-func checkForFlags(args []string) uint8 {
-	if len(args) == 1 {
-		printHelp()
-		return earlyReturn
-	}
-	for _, arg := range args {
-		if arg == "-h" || arg == "--help" {
-			printHelp()
-			return earlyReturn
-		}
-	}
-	return continueProgram
-}
-
 var (
 	pythonPath = os.Getenv("HOME") + "/.local/share/gosplat/src/python_helper/fast_model_compare.py"
 	modelBin   = os.Getenv("HOME") + "/.local/share/gosplat/fast-fb-model.bin"
 )
 
 const (
-	earlyReturn     uint8 = 1
-	continueProgram uint8 = 2
+	earlyReturn      uint8 = 1
+	continueProgram  uint8 = 2
+	accuracyOn       uint8 = 3
+	nameSuggestionOn uint8 = 4
 )
 
-func main() {
-	if checkForFlags(os.Args) == earlyReturn {
-		return
-	}
-	dir := os.Args[1]
+// flagValues
+var (
+	dir             string
+	accuracy        int
+	help            bool
+	nameSuggestions bool
+)
+
+func checkForFlags() {
+	flag.StringVar(&dir, "d", ".", "Directory; Dir which analysis starts on")
+	flag.IntVar(&accuracy, "a", 1, "Accuracy; The amount of positives it reports on (not it will increase false positives)")
+	flag.BoolVar(&nameSuggestions, "ns", false, "Name suggestions; if positive suggest a better name for package")
```

Comment:
Do you mean `If true, suggest a better name for a package` ? And not positive?

Or even better should it not be a flag that takes no argument or value like `cp -f ` or similar? That just enables it if flag is present.

Reply:
Look at flag documentation for flag.BoolVar

Reply:
Is there a better option than boolFlag? Otherwise let it be.